src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

```diff
@@ -2010,23 +2137,290 @@ private void resetVariable() {
 		inventoryRowListAdded.clear();
 		lotsSaved.clear();
 	}
-	
+
 	private boolean checkParameters(String wardCode, String chargeCode, String dischargeCode, Integer suplierId, String reference, LocalDateTime date) {
 		if (!lotsSaved.isEmpty() || !inventoryRowListAdded.isEmpty() || !lotsDeleted.isEmpty() || !inventoryRowsToDelete.isEmpty()
-							|| (destination != null && !destination.getCode().equals(wardCode))
-							|| (chargeType != null && !chargeType.getCode().equals(chargeCode))
-							|| (dischargeType != null && !dischargeType.getCode().equals(dischargeCode))
-							|| (supplier != null && !supplier.getSupId().equals(suplierId)) || (destination == null && wardCode != null)
-							|| (chargeType == null && chargeCode != null) || (dischargeType == null && dischargeCode != null)
-							|| (supplier == null && suplierId != null) || (reference != null && !reference.equals(newReference))
-							|| !date.toLocalDate().equals(dateInventory.toLocalDate())) {
+						|| (destination != null && !destination.getCode().equals(wardCode))
+						|| (chargeType != null && !chargeType.getCode().equals(chargeCode))
+						|| (dischargeType != null && !dischargeType.getCode().equals(dischargeCode))
+						|| (supplier != null && !supplier.getSupId().equals(suplierId)) || (destination == null && wardCode != null)
+						|| (chargeType == null && chargeCode != null) || (dischargeType == null && dischargeCode != null)
+						|| (supplier == null && suplierId != null) || (reference != null && !reference.equals(newReference))
+						|| !date.toLocalDate().equals(dateInventory.toLocalDate())) {
 			return true;
 		}
 		return false;
 	}
+
+	private JComboBox<MedicalType> getJComboMedicalType() {
+		if (medicalTypeComboBox == null) {
+			medicalTypeComboBox = new JComboBox<>();
+			try {
+				List<MedicalType> medicalTypes = medicalTypeManager.getMedicalType();
+				MedicalType medicalType = new MedicalType(MessageBundle.getMessage("angal.common.all.txt"), MessageBundle.getMessage("angal.common.all.txt"));
+				medicalTypeComboBox.addItem(medicalType);
+				for (MedicalType medType : medicalTypes) {
+					medicalTypeComboBox.addItem(medType);
+				}
+			} catch (OHServiceException e) {
+				OHServiceExceptionUtil.showMessages(e);
+			}
+
+			medicalTypeComboBox.addActionListener(actionEvent -> {
+				medicalTypeSelected = (MedicalType) medicalTypeComboBox.getSelectedItem();
+			});
+		}
+		return medicalTypeComboBox;
+	}
+
+	private JRadioButton getAllRadioButton() {
+		if (radioButtonAll == null) {
+			radioButtonAll = new JRadioButton(MessageBundle.getMessage("angal.common.all.btn"));
+			radioButtonAll.setMnemonic(MessageBundle.getMnemonic("angal.common.all.btn.key"));
+			radioButtonAll.setSelected(true);
+		}
+		radioButtonAll.addActionListener(actionEvent -> {
+			if (radioButtonAll.isSelected()) {
+				radioOnlyNonZero.setSelected(false);
+				radioWithMovement.setSelected(false);
+			}
+		});
```

Comment:
In java swing this logic can be avoided by adding the three `JRadioButtons` to the same `ButtonGroup`

Reply:
okay I have remove the that